### ISSUE TITLE
NETSCRIPT: hackChance(hacknetServer, player) throws error

### ISSUE THREAD

**Issue by bupjae** [bitburner-official/bitburner-src#281]
Version of game: both Bitburner 2.1.0 (0133945c) and Bitburner 2.2.0 (313f6ada)

Step to reproduce:
1. Get SF9.1 or enter inside BN9
2. Prepare Formulas.exe
3. Buy one hacknet server
4. Write and run following NS2 script
```javascript
/** @param {NS} ns */
export async function main(ns) {
	const server = ns.getServer('hacknet-node-0');
	const player = ns.getPlayer();
	const hackPercent = ns.formulas.hacking.hackPercent(server, player);
	ns.tprint(hackPercent);
}
```

Expected result: some sane value. 0 is acceptable

Actual result: Type error is thrown
`formulas.hacking.hackPercent: server should be a Server.`

The prototype of `getServer` is `getServer(host?: string): Server;`, returning Server object.
The prototype of `hackPercent` is `hackPercent(server: Server, player: Person): number;`, accepting Server object.
According to prototype, `hackPercent` shouldn't throw type error even if server is actually hacknet server.

**Comment by Snarling** [comment]
Yeah it is a typing issue. Internally HacknetServer is a different type from regular Server (which is basically a hackable-server).

You'd still get an error with your usage there, because you can't hack hacknet servers, but the error is wrong so I'm keeping this issue open.

**Comment by Snarling** [comment]
Fixed by 0442f3d
